From bunk@stusta.de Mon Mar 19 02:26:29 2007
Date: Mon, 19 Mar 2007 10:26:32 +0100
From: Adrian Bunk <bunk@stusta.de>
To: "Jeremy A. Roberson" <jroberson@gtcocalcomp.com>, Greg Kroah-Hartman <gregkh@suse.de>
Subject: USB: gtco.c: fix a use-before-check
Message-ID: <20070319092632.GQ752@stusta.de>

NULL checks should be before the first dereference.

Spotted by the Coverity checker.

Signed-off-by: Adrian Bunk <bunk@stusta.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/input/gtco.c |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

--- a/drivers/usb/input/gtco.c
+++ b/drivers/usb/input/gtco.c
@@ -1047,13 +1047,10 @@ static void gtco_disconnect(struct usb_i
 
 	/* Grab private device ptr */
 	struct gtco    *device = usb_get_intfdata (interface);
-	struct input_dev *inputdev;
-
-	inputdev = device->inputdevice;
 
 	/* Now reverse all the registration stuff */
 	if (device) {
-		input_unregister_device(inputdev);
+		input_unregister_device(device->inputdevice);
 		usb_kill_urb(device->urbinfo);
 		usb_free_urb(device->urbinfo);
 		usb_buffer_free(device->usbdev, REPORT_MAX_SIZE,
